api/scpca_portal/models/original_file.py

```diff
@@ -73,20 +77,37 @@ def get_from_dict(cls, file_object, bucket, sync_timestamp):
             project_id=s3_key_info.project_id,
             sample_id=s3_key_info.sample_id,
             library_id=s3_key_info.library_id,
-            is_single_cell=s3_key_info.is_single_cell,
-            is_spatial=s3_key_info.is_spatial,
-            is_cite_seq=s3_key_info.is_cite_seq,
-            is_bulk=s3_key_info.is_bulk,
-            is_single_cell_experiment=s3_key_info.is_single_cell_experiment,
-            is_anndata=s3_key_info.is_anndata,
-            is_metadata=s3_key_info.is_metadata,
+            is_single_cell=(Modalities.SINGLE_CELL in modalities),
+            is_spatial=(Modalities.SPATIAL in modalities),
+            is_cite_seq=(Modalities.CITE_SEQ in modalities),
+            is_bulk=(Modalities.BULK_RNA_SEQ in modalities),
+            format=format,
+            is_single_cell_experiment=(format == FileFormats.SINGLE_CELL_EXPERIMENT),
+            is_anndata=(format == FileFormats.ANN_DATA),
+            is_supplementary=(format == FileFormats.SUPPLEMENTARY),
+            is_metadata=(format == FileFormats.METADATA),
             is_merged=s3_key_info.is_merged,
             is_project_file=s3_key_info.is_project_file,
-            is_downloadable=s3_key_info.is_downloadable,
+            is_downloadable=OriginalFile._is_downloadable(modalities, format),
         )
 
         return original_file
 
+    @staticmethod
+    def _is_downloadable(modalities, format):
```

Comment:
Does it make more sense to just pass s3_key_info? I think we should type hint this method as well.